src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -454,22 +454,26 @@ private function darwinRealpath($real)
             if (isset(self::$darwinCache[$kDir])) {
                 $real = self::$darwinCache[$kDir][0];
             } else {
-                $dir = getcwd();
-                chdir($real);
-                $real = getcwd().'/';
-                chdir($dir);
-
-                $dir = $real;
-                $k = $kDir;
-                $i = \strlen($dir) - 1;
-                while (!isset(self::$darwinCache[$k])) {
-                    self::$darwinCache[$k] = [$dir, []];
-                    self::$darwinCache[$dir] = &self::$darwinCache[$k];
-
-                    while ('/' !== $dir[--$i]) {
+                try {
+                    $dir = getcwd();
+                    chdir($real);
+                    $real = getcwd().'/';
+                    chdir($dir);
+
+                    $dir = $real;
+                    $k = $kDir;
+                    $i = \strlen($dir) - 1;
+                    while (!isset(self::$darwinCache[$k])) {
+                        self::$darwinCache[$k] = [$dir, []];
+                        self::$darwinCache[$dir] = &self::$darwinCache[$k];
+
+                        while ('/' !== $dir[--$i]) {
+                        }
+                        $k = substr($k, 0, ++$i);
+                        $dir = substr($dir, 0, $i--);
                     }
-                    $k = substr($k, 0, ++$i);
-                    $dir = substr($dir, 0, $i--);
+                } catch (\ErrorException $exception) {
```

Comment:
Catching and discarding all kinds of `ErrorException` hides potential problems. We shouldn‘t do this.